src/api/spec/db/data/regenerate_notifications_spec.rb

```diff
@@ -0,0 +1,183 @@
+require 'rails_helper'
+require Rails.root.join('db/data/20200326221616_regenerate_notifications.rb')
+
+RSpec.describe RegenerateNotifications, type: :migration do
+  describe 'up' do
+    let(:owner) { create(:confirmed_user, login: 'bob') }
+    let(:requester) { create(:confirmed_user, login: 'ann') }
+    let(:project) { create(:project, name: 'bob_project', maintainer: [owner]) }
+    let(:package) { create(:package, name: 'bob_package', project: project) }
+    let(:another_package) { create(:package) }
+
+    let!(:new_bs_request) do
+      create(:bs_request_with_submit_action,
+             state: :new,
+             creator: requester,
+             target_project: project,
+             target_package: package,
+             source_package: another_package)
+    end
+    let!(:declined_bs_request) do
+      bs_request_to_decline =
+        create(:bs_request_with_submit_action,
+               state: :new,
+               creator: requester,
+               target_project: project,
+               target_package: package,
+               source_package: another_package)
+      bs_request_to_decline.update_attributes(state: 'declined')
+      bs_request_to_decline
+    end
+    let!(:old_declined_bs_request) do
+      bs_request_to_decline =
+        create(:bs_request_with_submit_action,
+               state: :new,
+               creator: requester,
+               target_project: project,
+               target_package: package,
+               source_package: another_package,
+               created_at: 101.days.ago)
+      bs_request_to_decline.update_attributes(state: 'declined')
+      bs_request_to_decline
+    end
+    let!(:revoked_bs_request) { create(:bs_request, type: 'maintenance_release', state: :revoked) } # This shouldn't regenerate notification
+
+    subject { RegenerateNotifications.new.up }
+
+    context 'for RequestCreate Notifications' do
+      let!(:subscription) { create(:event_subscription_request_created, receiver_role: 'target_maintainer', user: owner) }
+
+      before do
```

Comment:
You can use `subject!` above and remove this `before` block, as described [here](https://relishapp.com/rspec/rspec-core/docs/subject/explicit-subject#use-the-%60subject!%60-bang-method-to-call-the-definition-block-before-the-example)

Reply:
As we discussed offline, this is not possible because some `let` are executed after the subject (job) is call so the tests break.

Reply:
Agree